### ISSUE TITLE
PyPlot indexing is 0-based

### ISSUE THREAD

**Issue by zoohair** [stevengj/PyPlot.jl#117]
It seems like the Julia convention is to start indexing at 1, but PyPlot indexes at 0. I imagine this is likely due to the fact that python itself is 0-based.

Without going into a 0 vs 1 base indexing discussion, it might be worth considering if there's a way to make PyPlot wrapper behavior consistent with Julia. 

Otherwise, you get oddities like this:
`PyPlot.plot(1:10)` results in a line going through (0,1)

In contrast, in Matlab doing `plot(1:10)` results in a line going through (1,1)

**Comment by stevengj** [comment]
I would say that the choice of origin `PyPlot.plot(1:10)` is fairly arbitrary to begin with, and is usually employed when you don't care too much about the x axis.   If you need a specific x axis you should usually pass it explicitly; I'm skeptical that it's worth changing the matplotlib defaults (although it could certainly be done by overloading `plot`).

**Comment by zoohair** [comment]
I agree that it might not be worth changing the matplotlib defaults. However, I wouldn't say that the choice of indexing is arbitrary: when only one axis is given to the plot function, it is often implied that the x-axis is the index of the elements. For a language that is 0-based indexing such as python, that would be 0:(n-1), for a language that is 1-based indexing that should be 1:n.

Both matplotlib and matlab mention something about this in their help:
Matlab: `plot(Y) creates a 2-D line plot of the data in Y versus the index of each value.`
matplotlib : `plot(y)           # plot y using x as index array 0..N-1`

I came across this because in one case I manually specified the x-axis because my y data was not sorted, and in the other I did not because my y-data was sorted by index. My plots were not making sense and when I dug deeper I realized that the default behavior was inconsistent with the 1-based indexing of Julia.

Feel free to mark as non-issue.
